File: sharedModule/src/main/java/ru/nekit/android/qls/pupil/avatar/PupilAvatarViewBuilder.java
package ru.nekit.android.qls.pupil.avatar;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.AppCompatImageView;
import android.view.ViewGroup;

import java.util.List;

import ru.nekit.android.qls.pupil.Pupil;

public class PupilAvatarViewBuilder {

    public static void build(@NonNull Context context, @NonNull Pupil pupil,
                             @NonNull ViewGroup pupilAvatarContainer) {
        List<PupilAvatarPartAndVariant> pupilAvatarPartAndVariants =
                PupilAvatarConverter.toPupilAvatarPartAndVariant(pupil);
        for (PupilAvatarPartAndVariant pupilAvatarPartAndVariant : pupilAvatarPartAndVariants) {
            AppCompatImageView avatarPartImageRepresentation =
                    new AppCompatImageView(context);
            avatarPartImageRepresentation.setImageResource(
                    pupilAvatarPartAndVariant.pupilAvatarPart.getVariant(pupilAvatarPartAndVariant.variant));
            pupilAvatarContainer.addView(avatarPartImageRepresentation);
        }
    }
}

File: sharedModule/src/main/java/ru/nekit/android/qls/pupil/avatar/PupilAvatarViewBuilder.kt
package ru.nekit.android.qls.pupil.avatar

import android.content.Context
import android.support.v7.widget.AppCompatImageView
import android.view.ViewGroup
import ru.nekit.android.qls.shared.model.Pupil

object PupilAvatarViewBuilder {

    fun build(context: Context, pupil: Pupil, pupilAvatarContainer: ViewGroup) {
        val pupilAvatarPartAndVariants = PupilAvatarConverter.toPupilAvatarPartAndVariant(pupil)
        for (pupilAvatarPartAndVariant in pupilAvatarPartAndVariants) {
            val avatarPartImageRepresentation = AppCompatImageView(context)
            avatarPartImageRepresentation.setImageResource(
                    pupilAvatarPartAndVariant.pupilAvatarPart.variants[pupilAvatarPartAndVariant.variant])
            pupilAvatarContainer.addView(avatarPartImageRepresentation)
        }
    }
}
